### PR TITLE
Update params dependencies into value input channels #2982

### PR DESCRIPTION
**Pull request by vucka**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

**Comment by maxulysse** [comment]
Hi @vucka can you rename the title, I feel like it doesn't correspond to what is actually happening in the PR

**Comment by maxulysse** [comment]
@vucka any chance you can try and fix the merge conflicts?

File: subworkflows/nf-core/bam_rseqc/main.nf [hunk 1]
```diff
@@ -127,42 +127,42 @@ workflow BAM_RSEQC {
...
 
-    innerdistance_distance          // channel: [ val(meta), txt ]
-    innerdistance_freq              // channel: [ val(meta), txt ]
-    innerdistance_mean              // channel: [ val(meta), txt ]
-    innerdistance_pdf               // channel: [ val(meta), pdf ]
-    innerdistance_rscript           // channel: [ val(meta), r   ]
+    innerdistance_distance          // channel: [ val(meta), path(txt) ]
+    innerdistance_freq              // channel: [ val(meta), path(txt) ]
+    innerdistance_mean              // channel: [ val(meta), path(txt) ]
+    innerdistance_pdf               // channel: [ val(meta), ppath(pdf) ]
```
Comment (LouisLeNezet) [inline subworkflows/nf-core/bam_rseqc/main.nf:142]: Should'nt it be path()

File: subworkflows/nf-core/bcl_demultiplex/main.nf [hunk 2]
```diff
@@ -10,7 +10,7 @@ include { UNTAR      } from "../../../modules/nf-core/untar/main"
 
 workflow BCL_DEMULTIPLEX {
     take:
-        ch_flowcell     // [[id:"", lane:""],samplesheet.csv, path/to/bcl/files]
+        ch_flowcell      // channel: [ [id:"", lane:""], path(samplesheet), path(files) ]     (path to samplesheet.csv and path to bcl files)
```
Comment (LouisLeNezet) [inline subworkflows/nf-core/bcl_demultiplex/main.nf:13]: There is an extra space.
Also do we need the string inside the parentheses ? As we never specified it before.

File: subworkflows/nf-core/fasta_index_dna/main.nf [hunk 3]
```diff
@@ -13,9 +13,9 @@ include { SNAPALIGNER_INDEX as SNAP_INDEX   } from '../../../modules/nf-core/sna
 workflow FASTA_INDEX_DNA {
 
     take:
-        ch_fasta        // channel: [mandatory] [ val(meta), path(fasta) ]
-        ch_altliftover  // channel: [mandatory, if aligner is bwamem or bwamem2 or snap] [ val(meta), path(altliftover) ]
-        val_aligner     // string:  [mandatory] aligner [bowtie2, bwamem, bwamem2, dragmap, snap]
+        ch_fasta        // channel:  [ val(meta), path(fasta) ]    (mandatory)
+        ch_altliftover  // channel:  [ val(meta), path(altliftover) ]    (mandatory, if aligner is bwamem or bwamem2 or snap)
+        val_aligner     // string:  aligner [bowtie2, bwamem, bwamem2, dragmap, snap] (mandatory)
```
Comment (LouisLeNezet) [inline subworkflows/nf-core/fasta_index_dna/main.nf:18]: Maybe `aligner` is not necessary.

File: subworkflows/nf-core/fastq_align_bowtie2/main.nf [hunk 4]
```diff
@@ -7,11 +7,11 @@ include { BAM_SORT_STATS_SAMTOOLS } from '../bam_sort_stats_samtools/main'
 
 workflow FASTQ_ALIGN_BOWTIE2 {
     take:
-    ch_reads          // channel: [ val(meta), [ reads ] ]
-    ch_index          // channel: /path/to/bowtie2/index/
-    save_unaligned    // val
-    sort_bam          // val
-    ch_fasta          // channel: /path/to/reference.fasta
+    ch_reads            // channel: [ val(meta), path(reads) ]
+    ch_index            // channel: [ path(index) ]     (/path/to/bowtie2/index/)
```
Comment (LouisLeNezet) [inline subworkflows/nf-core/fastq_align_bowtie2/main.nf:11]: Do we really need what's in the parentheses ?
If we want to standardize those comment, maybe taking this off would be better.
I also don't think some of them are really useful `(/path/to/reference.fasta)`

File: subworkflows/nf-core/fastq_align_chromap/main.nf [hunk 5]
```diff
@@ -31,11 +31,12 @@ workflow FASTQ_ALIGN_CHROMAP {
...
-    flagstat          = BAM_SORT_STATS_SAMTOOLS.out.flagstat // channel: [ val(meta), [ flagstat ] ]
-    idxstats          = BAM_SORT_STATS_SAMTOOLS.out.idxstats // channel: [ val(meta), [ idxstats ] ]
+    bam               = BAM_SORT_STATS_SAMTOOLS.out.bam      // channel: [ val(meta), path(bam) ]
+    bai               = BAM_SORT_STATS_SAMTOOLS.out.bai      // channel: [ val(meta), path(bai) ]
+    stats             = BAM_SORT_STATS_SAMTOOLS.out.stats    // channel: [ val(meta), path(stats) ]
+    flagstat          = BAM_SORT_STATS_SAMTOOLS.out.flagstat // channel: [ val(meta), path(flagstat) ]
+    idxstats          = BAM_SORT_STATS_SAMTOOLS.out.idxstats // channel: [ val(meta), path(idxstats) ]
 
-    versions          = ch_versions                          //    path: versions.yml
+    versions          = ch_versions                          // channel: [(path: versions.yml)]
```
Comment (LouisLeNezet) [inline subworkflows/nf-core/fastq_align_chromap/main.nf:40]: Should be `path(versions.yml)`

File: subworkflows/nf-core/fastq_align_dna/main.nf [hunk 6]
```diff
@@ -15,8 +15,8 @@ include { SNAPALIGNER_ALIGN as SNAP_ALIGN   } from '../../../modules/nf-core/sna
 
 workflow FASTQ_ALIGN_DNA {
     take:
-        ch_reads            // channel: [mandatory] meta, reads
-        ch_aligner_index    // channel: [mandatory] aligner index
+        ch_reads            // channel: [mandatory] [ val(meta), path(reads)]
```
Comment (LouisLeNezet) [inline subworkflows/nf-core/fastq_align_dna/main.nf:18]: If it's for the sake of normalizing, do you also want to standardize the mandatory comment.
Here it is added before the description but in fasta_dna_index you put it after and inside parentheses.

File: subworkflows/nf-core/fastq_contam_seqtk_kraken/main.nf [hunk 7]
```diff
@@ -9,9 +9,9 @@ include { SEQTK_SAMPLE               } from '../../../modules/nf-core/seqtk/samp
 workflow FASTQ_CONTAM_SEQTK_KRAKEN {
 
     take:
-        ch_reads    //channel: [mandatory] meta,reads
+        ch_reads    //channel: [mandatory] [ val(meta), path(reads)]
         sample_size //string:  [mandatory] number of reads to subsample
-        kraken2_db  //string:  [mandatory] path to Kraken2 DB to use for screening
+        kraken2_db  //channel: [mandatory] [ path(db) ]     path to Kraken2 DB to use for screening)
```
Comment (LouisLeNezet) [inline subworkflows/nf-core/fastq_contam_seqtk_kraken/main.nf:14]: Missing space and parenthese

File: subworkflows/nf-core/fastq_fastqc_umitools_trimgalore/main.nf [hunk 8]
```diff
@@ -36,28 +36,28 @@ workflow FASTQ_FASTQC_UMITOOLS_TRIMGALORE {
...
             // Discard R1 / R2 if required
             if (umi_discard_read in [1,2]) {
                 UMITOOLS_EXTRACT
                     .out
-                    .reads
+                    .ch_reads
                     .map {
-                        meta, reads ->
-                            meta.single_end ? [ meta, reads ] : [ meta + ['single_end': true], reads[umi_discard_read % 2] ]
+                        meta, ch_reads ->
```
Comment (LouisLeNezet) [inline subworkflows/nf-core/fastq_fastqc_umitools_trimgalore/main.nf:59]: I don't think you should change it here to ch_reads.
As you are inside a closure, I think this should be reads.

File: subworkflows/nf-core/fastq_subsample_fq_salmon/main.nf [hunk 9]
```diff
@@ -9,11 +9,11 @@ include { SALMON_QUANT } from '../../../modules/nf-core/salmon/quant/main'
...
-    ch_reads            // channel: [ val(meta), [ reads ] ]
-    ch_genome_fasta     // channel: /path/to/genome.fasta
-    ch_transcript_fasta // channel: /path/to/transcript.fasta
-    ch_gtf              // channel: /path/to/genome.gtf
-    ch_index            // channel: /path/to/salmon/index/
+    ch_reads            // channel: [ val(meta), path(reads) ]
+    ch_genome_fasta     // channel: [ path(g_fasta)]      (/path/to/genome.fasta)
+    ch_transcript_fasta // channel: [ path(t_fasta)]      (/path/to/transcript.fasta)
+    ch_gtf              // channel: [ path(gtf)]          (/path/to/genome.gtf)
+    ch_index            // channel: [ path(index)]                      (/path/to/salmon/index/)
```
Comment (LouisLeNezet) [inline subworkflows/nf-core/fastq_subsample_fq_salmon/main.nf:16]: If you keep the parentheses, you could also align them 😄

**Review by LouisLeNezet — COMMENTED** [review]
I really the work you've done to harmonize all those comment and variable.
Maybe we could also use this PR, to standardize also:
- `[mandatory]` or `(mandatory)`, before or after the description, maybe we could also not put them at all as most aren't added (maybe preferably add `(optional)`, they might be less to add)
- I think the `(/path/to/file.extension)` are not necessary, as the name of the channel and the channel description comment you added are sufficient.
- And as an open question to the nf-core maintainer: Should we let other type of comment on the input channel line ? (e.g. vcf_annotate_ensemblevep, vcf_gather_bcftools) ?

File: subworkflows/nf-core/bam_rseqc/main.nf (hunk 1 above)
Comment (Emiller88) [inline subworkflows/nf-core/bam_rseqc/main.nf:142]:
```suggestion
    innerdistance_pdf               // channel: [ val(meta), path(pdf) ]
```

**Review by Emiller88 — COMMENTED** [review]
Looks really good for me thanks for taking the time to do this!

@bentsherman When's that LSP with type hints coming? Or I need to go test out the latest version.

**Comment by SPPearce** [comment]
There are now 12 conflicted files, I'm struggling to see how to reconcile this and what the real changes actually are, so I'm minded to close this PR.

**Pull request closed without merging**